kdc/kerberos5.c

```diff
@@ -178,18 +178,12 @@ get_pa_etype_info2(krb5_context context,
 }
 
 static krb5_error_code
-set_salt_padata(METHOD_DATA *md, Salt *salt)
+set_salt_padata(krb5_context context,
```

Comment:
“This matches Windows behaviour” is a fine commit message for Samba, but not for Heimdal. Could you please explain what this fixes with reference to the relevant non-Microsoft specifications (e.g. RFC 4120).